bus/api.go

```diff
@@ -190,3 +190,11 @@ type MigrationContract struct {
 	HostIP  string               `json:"hostIP"`
 	ID      types.FileContractID `json:"id"`
 }
+
+// UploadParams contains the metadata needed by a worker to upload an object.
+type UploadParams struct {
+	CurrentHeight uint64
+	MinShards     uint8
+	TotalShards   uint8
+	Contracts     []Contract
```

Comment:
Might want to consider passing set names instead of contracts.

e.g. if you have renters set up in production and you load balance between a bunch of workers, none of them need to be aware of any contracts. You just give them the name of the set they should use and that's it. This is especially useful to keep unnecessary logic out of your load balancer which otherwise has to fetch the contracts first. You can just configure the set once.

Then the autopilot or some other service in the background can make sure that the set is up-to-date.

Reply:
Same goes for downloads btw. e.g. if you run renter behind an authentication service that creates JWTs for logged in users. Then the JWT could already contain the contract set to use for downloads and the service can attach that info to the user's request without any additional work for fetching contracts.